rd-net/RdFramework/Impl/MessageBroker.cs

```diff
@@ -74,8 +84,27 @@ private unsafe void Execute(IRdWireable reactive, byte[] msg)
             reactive.OnWireReceived(reader);
       }
     }
-    
-    
+
+    public void StartDeliveringMessages()
+    {
+      Assertion.Require(myIsQueueingAllMessages, "Already started delivering messages");
```

Comment:
The same as above, here we're touching `myIsQueueingAllMessages` without a lock which is supposed to guard it. Should we move it inside of the `lock` section?